algebras/algebra/src/main/scala/endpoints/algebra/Requests.scala

```diff
@@ -112,10 +112,36 @@ trait Requests extends Urls with Methods with SemigroupalSyntax {
 
   /**
     * Request with a `String` body.
+    *
+    *   - Server interpreters are expected to be permissive and accept multiple
+    *     content types (not just `text/plain`) using the charset defined in the
+    *     `Content-Type` header to decide how to decode the body. If no charset
+    *     is defined (or if there is no `Content-Type` header) the interpreters
+    *     are free to define their own defaults for decoding.
+    *
+    *   - Client interpreters produce an HTTP request with a `text/plain` content type.
+    *
+    * @see [[plainTextRequest]] if you want to avoid accepting any content type
     * @group operations
     */
   def textRequest: RequestEntity[String]
 
+  /**
+    * Request with content-type `text/plain`
+    *
+    *   - Server interpreters raise an error if the incoming request entity does
+    *     not have the right content type. By default, they produce a Bad Request (400)
+    *     response with a list of error messages in a JSON array. Refer to the documentation
+    *     of your server interpreter to customize this behavior.
+    *
+    *   - Client interpreters produce an HTTP request with a `text/plain` content type.
+    *
+    * @see [[textRequest]] for a more permissive text request entity.
+    *
+    * @group operations
+    */
+  def plainTextRequest: RequestEntity[String]
```

Comment:
Thanks a lot for clarifying the behavior of both operations.

Do we really need to have both `textRequest` and `plainTextRequest` in the library? It seems to me that we should pick one, and let library users add the other on their side, if they want to.

Reply:
If we were to pick one, I'd pick `plainTextRequest`. The deciding factor that convinced me to keep `textRequest` too is: I think there should still be a way to set up an endpoint that  accepts test requests that don't specify any content-type. `plainTextRequest` _requires_ `text/plain`, and that requirement is necessary for accurate OpenAPI documentation and `.orElse` for request entities.

If you agree that we want a way to accept test requests that don't specify content-type, the rest of the specification for `textRequest` follows pretty naturally.

Reply:
What do you think of just changing the existing behavior of `textRequest` to be more strict (and do the same for `jsonRequest`)? We can see later if it becomes really important to also have a more forgiving behavior for servers. I initially thought it could be a good default, but I think supporting #529 is more important so maybe it’s better to just change the default behavior.

Reply:
I'm in favour of such a change. Forcing content-types everywhere will mean that it is much harder to use #529 in a way that doesn't make sense.

But just to re-iterate: taking this path means requests without a content-type will always fail. Is that OK with you?

Reply:
Would it be possible to require a `Content-Type` _only_ if there are several possible request entity types, but not if there is only one?

Reply:
Yeah, I like that. It is probably going to complicate the servers a bit (they'll internally need to keep track of whether a `RequestEntity` is strict due to being part of an `orElse`), but that's fine.

Shall we do this? If so, I'm going to close this PR and open another to focus on `Content-Type` and the character encoding issues.

Reply:
> Shall we do this? If so, I'm going to close this PR and open another to focus on `Content-Type` and the character encoding issues.

Yes, that would be awesome! Thank you for the fruitful discussion.

Reply:
> Thank you for the fruitful discussion.

Likewise 😄